packages/xo-server/src/xapi-object-to-xo.js

```diff
@@ -260,6 +291,13 @@ const TRANSFORMS = {
       }
     })()
 
+    const {
+      managementAgentDetected,
+      pvDriversDetected,
+      pvDriversVersion,
+      pvDriversUpToDate,
+    } = getVmGuestToolsProps(obj)
```

Comment:
Why not inlining in the VM object?

Reply:
It'll complicate the implementation, better group the guest metrics context in a dedicated function.

Reply:
You could spread its returned value directly in the object.

Reply:
I did that specially for readability to avoid an indirection on searching provided properties but i can spread it if you see that it's useless.